src/imap/mod.rs

```diff
@@ -1366,14 +1366,17 @@ async fn precheck_imf(
             let delete_server_after = context.get_config_delete_server_after().await;
 
             if delete_server_after != Some(0) {
-                context
+                let moving = context
                     .do_heuristics_moves(server_folder.as_ref(), msg_id)
                     .await;
-                job::add(
-                    context,
-                    job::Job::new(Action::MarkseenMsgOnImap, msg_id.to_u32(), Params::new(), 0),
-                )
-                .await;
+
```

Comment:
```suggestion

// If the bcc-self message is not moved, directly add MarkSeen job, otherwise MarkSeen job is added after the Move Job completed. 
```